Support for `tuple_extract` instruction

taufique71 opened the pull request:
<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [Issue 78](https://github.com/themaplelab/swift/issues/78).

taufique71 committed:
mplementation complete for tuple_extract instruction

mmroz commented on include/swift/WALASupport/SILWalaInstructionVisitor.h:
```diff
@@ -24,136 +24,136 @@ class SILWalaInstructionVisitor : public SILInstructionVisitor<SILWalaInstructio
+  jobject visitSelectValueInst(SelectValueInst *SVI);
+  jobject visitSwitchEnumInst(SwitchEnumInst *SWI);
+  jobject visitTryApplyInst(TryApplyInst *TAI);
+  jobject visitTupleExtractInst(TupleExtractInst *TEI);
```
Not the correct place. Please use the same order as Apple.

taufique71 replied:
Can you specifically refer me to some ordering?

mmroz replied:
https://github.com/apple/swift/blob/master/docs/SIL.rst

mmroz commented on lib/WALASupport/SILWalaInstructionVisitor.cpp:
```diff
@@ -1558,4 +1558,56 @@ jobject SILWalaInstructionVisitor::visitTryApplyInst(TryApplyInst *TAI) {
+  auto fieldNo = TEI->getFieldNo();
+  
+  // We get the element from above two information
+  auto tupleTypeElt = tupleType->getElement(fieldNo);
```
Please avoid using auto as it gets confusing

mmroz commented on lib/WALASupport/SILWalaInstructionVisitor.cpp:
```diff
@@ -1558,4 +1558,56 @@ jobject SILWalaInstructionVisitor::visitTryApplyInst(TryApplyInst *TAI) {
\ No newline at end of file
+jobject SILWalaInstructionVisitor::visitTupleExtractInst(TupleExtractInst *TEI) {
+  // Operand is a tuple type from which some field needs to be extracted.
+  // So it must have been created earlier. We just find the corresponding nodee
```
Spelling mistake on ‘nodee’

mmroz commented on lib/WALASupport/SILWalaInstructionVisitor.cpp:
```diff
@@ -1558,4 +1558,56 @@ jobject SILWalaInstructionVisitor::visitTryApplyInst(TryApplyInst *TAI) {
+    string fieldTypeName = tupleTypeElt.getType().getString();
+    jobject fieldTypeNameNode = Wala->makeConstant(fieldTypeName.c_str());
+    fieldNode = Wala->makeNode(CAstWrapper::VAR, fieldTypeNameNode);
+  }
```
The fields are constants not VAR because tuples are immutable.
Please use field node set to empty node not nullptr

mmroz reviewed (commented):
We are also naming variables like MyVariable not using camelCasing (for better or worse) but we should remain consistent

taufique71 committed:
Made changes according to the PR review

- Changed the ordering of function declaration and definition
- Removed all the `auto` keywords and replaced them with appropriate type
  names
- Variable name casing changed from camel casing for keeping consistency
  with other contributions
- Fixed typo in comments
- Removed `VAR` type wrapping around constant nodes as tupe elements are
  not mutable

taufique71 commented:
Updated my implementation. @mmroz Can you review the changes again?

taufique71 commented:
@karimhamdanali Conflicts are resolved now

karimhamdanali reviewed (approved):
LGTM
